scripts/ioc_deploy.py

```diff
@@ -349,6 +389,33 @@ def main_perms(args: CliArgs) -> int:
     return rval
 
 
+def main_rebuild(args: CliArgs) -> int:
+    """
+    All main steps of the rebuild-only action.
+
+    This will be called when the rebuild subparser is included.
+
+    Will either return an int code or raise.
+    """
+    deploy_dir = get_local_target(args)
+    logger.info(f"Planning to rebuild {deploy_dir}")
+    if not args.auto_confirm:
+        user_text = input("Confirm target? yes/true or no/false\n")
+        if not is_yes(user_text, error_on_empty=False):
+            return ReturnCode.NO_CONFIRM
+    rval = _perms(deploy_dir=deploy_dir, allow_write=True, dry_run=args.dry_run)
+    if rval != ReturnCode.SUCCESS:
+        return rval
+    rval = make_in(deploy_dir=deploy_dir, dry_run=args.dry_run)
+    if rval != ReturnCode.SUCCESS:
+        logger.error(f"Nonzero return value {rval} from make")
+        return rval
+    rval = _perms(deploy_dir=deploy_dir, allow_write=False, dry_run=args.dry_run)
+    if rval == ReturnCode.SUCCESS:
+        logger.info("Rebuild complete!")
```

Comment:
Nit: So if we error out on this `_perms` call, we don't tell the user exactly what went wrong?  Even though we get loud in `main`, the message is a generic "deploy errored out"

Reply:
I was hoping that _perms itself would have interesting output and logging, I'll double-check

Reply:
I've confirmed that we get a clear error messages whenever we set the return code to non-success values.

Reply:
I'm going to refactor a bit more to not return return codes from subfunctions, raising is more useful and is handled in main appropriately